docs/guides/advanced/data/4844_data.md

```diff
@@ -0,0 +1,185 @@
+| Prepare for Circuit Input: | Ensure that `hash(vh \|\| x \|\| y)` is provided as a [public input for the circuit][scheduler_expect_4844], based on the [Schwarz-Zippel lemma][schwarz_zippel] - if 2 polynomials are equal in a random point, there are equal everywhere. |
+
+### Inside the state keeper
+
+| step                         | explanation                                            |
+| ---------------------------- | ------------------------------------------------------ |
+| Collect data                 | Get `c`, `x`, `y`, `blob_proof` from witness generator |
+| Create New Transaction Type: | Form a new blob transaction with a special `0x3` type. |
+
+### Inside the L1 contract
+
+| step                             | explanation                                                                                                                                                                                                              |
+| -------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| Receive Inputs:                  | Get `c`, `x`, `y`, and `blob_proof` as inputs from the [state keeper][executor_inputs].                                                                                                                                  |
+| Verification Hash:               | Obtain the verification hash through the BLOB_HASH precompile. [`_getBlobVersionedHash`][get_blob_vhash]                                                                                                                 |
+| Use Precompile for Verification: | Apply a [precompile][precompile_4844] to check if `p(x) == y` using the version hash, `c` and `blob_proof`. This is where the KZG scheme plays a crucial role, allowing the proof without needing the entire polynomial. |
+| Pass Public Input to Verifier    | Send `hash(vh \|\| x \|\| y)` as public input to the verifier using [`_batchAuxiliaryOutput`][executor_aux].                                                                                                             |
+
+This process simplifies how zkSync validates the attachment of 128kb of L2 data to L1, ensuring its correctness and
+integrity.
+
+### What, but why
+
+#### Why does prover need to compute the commitment
+
+Prover needs to come up with a 'random' point 'x' (otherwise Schwarz-Zippel lemma would not work), so it has to generate
+based on something. The simpler solution would be to generate it based on the hash of the pubdata, but unfortunately the
+L1 contract doesn't have access to the hash - so it would not be able to verify that the 'x' was really generated based
+on that hash.
+
+#### Why do I need to pass blob proof to L1 contract
```

Comment:
not sure if I understand. there are two proofs that need to be provided with the transaction:

- Blob proof (tbh, I didn't dive into why do we need it yet)
- The point evaluation proof (that `p(x) = y`). 

If we take the analogy of "merkle root", then I don't think i would need to only provide point evaluation proof (i.e. merkle path) and not anything else

Reply:
L1 contract only gets the opening proof, blob proof is used by nodes to ensure the blobs == kzg commitments == versioned hashes